scripts/data-retention-remove-user-data.py

```diff
@@ -53,17 +53,5 @@
         auth_token=get_auth_token('api', stage)
     )
     cutoff_date = datetime.now() - timedelta(days=365 * 3)
-
-    all_users = list(data_api_client.find_users_iter(personal_data_removed=False))
-
-    for user in all_users:
-        last_logged_in_at = datetime.strptime(user['loggedInAt'], '%Y-%m-%dT%H:%M:%S.%fZ')
-        if last_logged_in_at < cutoff_date and not user['personalDataRemoved']:
-            logger.warn(
-                f"{prefix}Removing personal data for user: {user['id']}"
-            )
-            if not dry_run:
-                data_api_client.remove_user_personal_data(
-                    user['id'],
-                    'Data Retention Script {}'.format(datetime.now().isoformat())
-                )
+    remove_supplier_data(data_api_client, logger, dry_run, cutoff_date)
```

Comment:
I am starting to have second thoughts about doing the declaration deletion in this script.

The current script needs to run nightly, as each day might have a new set of users that have logged on >3 years ago. But frameworks expire much more rarely than this - twice a year at most. So the declaration deletions could be run as a one-off script (covering both failed suppliers and for any frameworks expired >3 years ago). This would also reduce the risk of accidentally deleting 'live' declarations. Let's discuss when you get back @jonodrew.